src/registryDAO.mligo

```diff
@@ -181,12 +139,12 @@ let registry_DAO_decision_lambda (proposal, extras : proposal * contract_extra)
               update_receivers(current_set, receivers, (fun (i, c : address * address set) -> Set.remove i c))
         in (ops, Big_map.update "proposal_receivers" (Some (Bytes.pack new_set)) extras)
       end
-  | Configuration_proposal ->
-      let m_frozen_scale_value = unpack_nat_opt(find_map("frozen_scale_value", proposal.metadata)) in
-      let m_frozen_extra_value = unpack_nat_opt(find_map("frozen_extra_value", proposal.metadata)) in
-      let m_max_proposal_size = unpack_nat_opt(find_map("max_proposal_size", proposal.metadata)) in
-      let m_slash_scale_value = unpack_nat_opt(find_map("slash_scale_value", proposal.metadata)) in
-      let m_slash_division_value = unpack_nat_opt(find_map("slash_division_value", proposal.metadata)) in
+  | Configuration_proposal cp ->
+      let m_frozen_scale_value = cp.frozen_scale_value in
+      let m_frozen_extra_value = cp.frozen_extra_value in
+      let m_max_proposal_size = cp.max_proposal_size in
+      let m_slash_scale_value = cp.slash_scale_value in
+      let m_slash_division_value = cp.slash_division_value in
       let new_ce = match m_frozen_scale_value with
```

Comment:
The lines above are an unnecessary re-binding now, we can remove them and use the values directly, e.g. here:
```suggestion
      let new_ce = match cp.frozen_scale_value with
```